tests/testthat/test-cube-subset.R

```diff
@@ -299,6 +352,27 @@ test_that("[ method for cat array cubes", {
     expect_equal(as.array(subset_cube_withNA), as.array(cube_withNA)[, , 1:2])
 })
 
+
+test_that("subsetting with reordering works", {
+    cat_x_cat <- loadCube(test_path("cubes/cat-x-cat.json"))
+    expect_equal(as.array(cat_x_cat[c(2, 1), ]), as.array(cat_x_cat)[c(2, 1), ])
+    expect_equal(as.array(cat_x_cat[c(2, 1), c(2, 1)]), as.array(cat_x_cat)[c(2, 1), c(2, 1)])
+
+    catarray_x_mr <- loadCube(test_path("cubes/catarray-x-mr.json"))
+    expect_equal(
+        as.array(catarray_x_mr[c(2, 1), , ]),
+        as.array(catarray_x_mr)[c(2, 1), , ]
+        )
+    expect_equal(
+        as.array(catarray_x_mr[, , c(3, 1, 2)]),
+        as.array(catarray_x_mr)[, , c(3, 1, 2)]
+        )
+    expect_equal(
+        as.array(catarray_x_mr[, 5:2, ]),
+        as.array(catarray_x_mr)[, 5:2, ]
+        )
```

Comment:
Since these tests ought to be cheap, do you think we should add at least one with subsetting two dimensions at the same time? It would also be very good to test the useNA behavior is doing what we expect it to, and not accidentally leaking `usaNA = "always"`.

Reply:
I added a few more tests including some, I think the useNA behaviour is probably tested sufficiently with all the other cube-subset tests, but I added a few just in case.